mbf_abstract_nav/src/controller_action.cpp

```diff
@@ -0,0 +1,267 @@
+#include "mbf_abstract_nav/controller_action.h"
+
+namespace mbf_abstract_nav{
+
+
+ControllerAction::ControllerAction(
+    const std::string &action_name,
+    const RobotInformation &robot_info)
+    : AbstractAction(action_name, robot_info, boost::bind(&mbf_abstract_nav::ControllerAction::run, this, _1, _2))
+{
+}
+
+
+void ControllerAction::before(GoalHandle &goal_handle, AbstractControllerExecution& execution)
+{
+  if(threads_.size() == 1)
+  {
+    cmd_vel_pub_goal_id_ = goal_handle.getGoalID().id;
+    execution.enablePublishCmdVel();
+  }
+  else
+  {
+    execution.disablePublishCmdVel();
+  }
+}
+
+
+void ControllerAction::after(GoalHandle &goal_handle, AbstractControllerExecution& execution)
+{
+  if(goal_handle.getGoalID().id == cmd_vel_pub_goal_id_ && threads_.size() > 1)
+  {
+    execution.disablePublishCmdVel();
+    cmd_vel_pub_goal_id_ = goal_ids_.front();
+    AbstractControllerExecution::Ptr other_exec = executions_[cmd_vel_pub_goal_id_];
+    other_exec->enablePublishCmdVel();
```

Comment:
I can confirm changing to >= 1 fixes the issue, but then I get a crash here when sending three consecutive goal w/o waiting for any to finish. Looks like other_exec pointer is not valid

Reply:
As @spuetz suggested, skipping the last two commits fixes both issues